### PR TITLE
Pass variables to replace tokens in query

### LINKED ISSUE #679: tokens in Mapnik postgis sql sub-selects not replaced using options.variables for VectorTiles
Accordingly to documentation, mapnik.Map `render` method have `options` attribute that could contain `variables` object.

http://mapnik.org/documentation/node-mapnik/3.5/#Map.render

**options.variables**

> Mapnik 3.x ONLY: A javascript object containing key value pairs that should be passed into Mapnik as variables for rendering and for datasource queries. For example if you passed vtile.render(map,image,{ variables : {zoom:1} },cb) then the @zoom variable would be usable in Mapnik symbolizers like line-width:"@zoom" and as a token in Mapnik postgis sql sub-selects like (select \* from table where some_field > @zoom) as tmp (used when rendering a vector tile)

Also, example from `mapbox/tilelive-bridge` code:
https://github.com/mapbox/tilelive-bridge/blob/master/index.js#L281

``` js
var opts = {};
opts.variables = { "zoom_level": z };
map.render(vtile, opts, function(err, vtile) {
  ...
});
```
## Test case

I tried to pass variables to mapnik PostGIS SQL query accordingly to docs (**id = @fieldid**).

``` js
var mapnik = require('mapnik');
var fs = require('fs');

mapnik.register_default_input_plugins();
var map = new mapnik.Map(256, 256);
map.loadSync('./map_conf.xml');

var opts = {};
opts.variables = { "fieldid": 1 };

map.render(new mapnik.VectorTile(0, 0, 0), opts, function(err, vtile) {
  vtile.getData(function(err, pbf) {
      console.log(pbf);
  });
});
```

`map_conf.xml` file:

``` xml
<?xml version="1.0" encoding="utf-8"?>
<Map srs="+init=epsg:3857">
  <Layer name="field_shapes" status="on" srs="+init=epsg:4326">
    <Datasource>
      <Parameter name="type">postgis</Parameter>
      <Parameter name="host">localhost</Parameter>
      <Parameter name="dbname">db_development</Parameter>
      <Parameter name="user">db_user</Parameter>
      <Parameter name="password">db_pass</Parameter>
      <Parameter name="srid">4326</Parameter>
      <Parameter name="geometry_field">shape</Parameter>
      <Parameter name="table">
  (SELECT id, shape FROM fields WHERE id = @fieldid) as field_shapes
      </Parameter>
      <Parameter name="estimate_extent">false</Parameter>
      <Parameter name="extent">-150,-85,150,85</Parameter>
    </Datasource>
  </Layer>
</Map>
```
## Problem

`id = @fieldid` must be replaced with `id = 1`, but I always get `id = null`.

From log of SQL queries:

``` sql
SELECT ST_AsBinary("shape") AS geom,"id" FROM
  (SELECT id, shape FROM fields WHERE id = null) as field_shapes
WHERE "shape" && ST_SetSRID('BOX3D(-150 -85,150 85)'::box3d, 4326)
```
## Mapnik version

``` js
> mapnik
...
version: '3.5.13',
versions: 
   { node: '4.4.3',
     v8: '4.5.103.37',
     boost: '1.59.0',
     boost_number: 105900,
     mapnik: '3.0.11',
     mapnik_number: 300011,
     mapnik_git_describe: 'v3.0.11',
     cairo: '1.14.4' },
...
```

### PR DESCRIPTION
**Pull request by rafatower**
This PR fixes #679. It comprises the following changes:
- the patch in https://github.com/CartoDB/node-mapnik/pull/5/commits/5182b7168f3c460cf19b94d98e9cfb9f9a4512f2 cherry-picked on master
- an end-to-end test for it
- the dependency `mapnik-vector-tile` updated to current master commit (to be changed).

Some remarks:
- As the change is built on https://github.com/mapbox/mapnik-vector-tile/pull/248 and takes advantage of the optional second argument of the vector tile processor constructor, updating to a newer version of `mapnik-vector-tile` is required, otherwise it will fail at build time. @springmeyer do you plan to release a `1.4.1` or `1.5.0` version anytime soon?
- The test depends on having postgres+postgis installed. Otherwise it is skipped. I'd like to add it to CI builds, if possible.

Comments & suggestions welcomed.

### CODE REVIEW

**Comment by springmeyer** [comment]
Noticed this is failing currently due to unrelated regression to ndoe v0.10.x support. Please rebase from master once https://github.com/mapnik/node-mapnik/pull/810 is merged.

> @springmeyer do you plan to release a 1.4.1 or 1.5.0 version anytime soon?

mapnik-vector-tile@1.5.0 is now published.

> The test depends on having postgres+postgis installed. Otherwise it is skipped. I'd like to add it to CI builds, if possible.

👍  And ideally on OSX as well as Travis. As you know postgres/postgis is a challenging install. I would prefer not to depend on third-party launchpad PPAs that change over time and may break unpredictably. For this reason I would prefer postgres/postgis is installed from mason, which I can update if needed to keep node-mapnik tests passing against different versions. Want to have a look at https://github.com/springmeyer/postgis-mason-install for an example of how to install this on travis? Or provide an alternative strategy?

**Comment by rafatower** [comment]
I rebased from master, updated `mapnik-vector-tile@1.5.0` (thanks for the prompt release BTW), added the scripts to install postgis from mason and integrated them with travis build. Hope this is now ready to go. Thanks!

File: test/vector-tile.postgis.test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                });
+        } else {
+            console.warn('postgis input datasource not registered.');
+            return done();
+        }
+    });
+
+    it('passes variables to replace tokens in query', function(done) {
+        if (!hasPostgisAvailable) {
+            this.skip('postgis not availabel');
```
Comment (springmeyer) [inline test/vector-tile.postgis.test.js]: Minor spelling fix needed: `availabel` -> `available`.

**Comment by springmeyer** [comment]
@rafatower thanks for putting together the testing quickly. Looks great. I am headed on vacation for a week, but I will plan to test this myself and merge when back.

**Comment by springmeyer** [comment]
> I will plan to test this myself and merge when back.

Sorry I've not gotten to this. I've become too busy with other responsibitilies to dive into this code with the attention I feel is needed. This is a very cool feature, but needs testing to ensure it does not break anything in an unintended way. Is there anyone else reading this could provide:

 - a second set of eyes?
 - test the feature in their downstream application to ensure it works?
 - help ensure all the tests are passing on travis with this PR?

**Comment by springmeyer** [comment]
Going to merge this into master without testing further. This is a rare exception for me, but the change is primarily tests so this should be safe. Please ping here if any hits any problems due to this change and we'll investigate.

**Pull request merged**